tests/unit/generateExpiryDate.unitTest.ts

```diff
@@ -106,10 +108,11 @@ describe("TestResultsService calling generateExpiryDate", () => {
                                     }
                                 };
                             });
-                            testResultsService = new TestResultsService(new MockTestResultsDAO());
 
                             const expectedExpiryDate = new Date("2021-01-31");
-                            return testResultsService.generateExpiryDate(psvTestResult)
+                            vehicleTestController.dataProvider.testResultsDAO = new MockTestResultsDAO();
```

Comment:
Wouldn't it be easier to do this call once in beforeEach() ?

Reply:
It won't work because beforeEach is happening before assigning the implementation to MockTestResultDAO.